state-chain/pallets/cf-threshold-signature/src/lib.rs

```diff
@@ -364,6 +363,8 @@ pub mod pallet {
 		/// A marker trait identifying the chain that we are signing for.
 		type TargetChainCrypto: ChainCrypto;
 
+		type TargetChain: Chain;
+
```

Comment:
```suggestion
		/// A marker trait identifying the chain that we are signing for.
		type TargetChainCrypto: ChainCrypto;

		type TargetChain: Chain<ChainCrypto = Self::TargetChainCrypto>;

```

Just to make sure can't do something stupid...

Reply:
Actually I'm not sure we need (or *can* provide) the target chain here. Multiple chains can use the same instance of this pallet.

Reply:
that's fine, though. If it is the same pallet, then it will also have the same types and the same serialization. All we need is for different instances of this generic pallet to have different names. The most straightforward way would be to use the NAME parameter of the 'Chain' type (ChainCrypto does not provide a name)

Reply:
Then we should add a name? Otherwise it's just confusing. We will see 'Arbitrum' even when it's Ethereum. If we're going to add names, we might as well make them accurate?